### PR TITLE
Feature/ab cleanup

### PR DESCRIPTION
**Pull request by mjoyce91**
Merge https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/506 first

Removes a bidder from the Available Bidders list if their handshake is registered.

### CODE REVIEW

File: talentmap_api/fsbid/views/cdo.py [hunk 1]
```diff
@@ -109,10 +110,17 @@ def put(self, request, pk, client_id):
...
+            # Generate a notification
             Notification.objects.create(owner=owner,
                                         tags=['bidding'],
                                         message=message)
             send_email(subject=message, body='Navigate to TalentMAP to see your updated bid tracker.', recipients=[owner.user.email])
+
+            # Remove the bidder from the available bidders list
+            ab = AvailableBidders.objects.filter(bidder_perdet=client_id)
+            if ab.exists():
+                ab.first().delete()
```
Comment (scott062) [inline talentmap_api/fsbid/views/cdo.py:122]: We've never formally addressed this, but we may want to archive this instead with the AB archive column. However, I can't remember if WS is using that archive column or not - need to follow up with Vijay. Do you prefer just to delete to keep it as clean as possible until we implement something that would "unarchive" someone?

Comment (mjoyce91) [reply talentmap_api/fsbid/views/cdo.py:122]: @scott062 - I guess my thought here was that this person would likely not be added to the AB list again for multiple years - all of the data that we stored about them might not be relevant if they were added again in the future - and could even be confusing if they got added in the future and the CDO didn't review the old comments/bureau/reason etc. I think of the archiving as an important feature when you add/remove someone within a short timespan.

Comment (scott062) [reply talentmap_api/fsbid/views/cdo.py:122]: Agreed - Makes sense!

**Comment by scott062** [comment]
Inner `try / except` causes the code to exit prematurely on except preventing the correct removal from AB list

**Comment by mjoyce91** [comment]
@scott062 updated to use `finally`

**Pull request merged**